### PR TITLE
[SYCLomatic #1532] Add test for user option `--sycl-file-extension`

### PR DESCRIPTION
**Pull request by the-slow-one**
A battery of tests is added which checks the following behavoir
* invoke dpct without the `--sycl-file-extension`
* invoke dpct with the `--sycl-file-extension` and options `dp-cpp`, `sycl-cpp`, and `cpp`.
* invoke dpct with incorrect value to `--sycl-file-extension`.

### CODE REVIEW

**Review by AndyCHHuang — APPROVED** [review]
LGTM

File: behavior_tests/src/user-defined-sycl-file-extension/do_test.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    of migrated files with correct extension
+    """
+    sub_command = f"{test_config.CT_TOOL} main.cu --out-root {{0}}"
+
+    # commands to test three options to --sycl-file-extension and the defualt
+    # behaviour
+    cmds_n_results = {
+        f"{sub_command} --sycl-file-extension=sycl-cpp": "main.sycl.cpp",
+        f"{sub_command} --sycl-file-extension=dp-cpp": "main.dp.cpp",
+        f"{sub_command} --sycl-file-extension=cpp": "main.cpp",
```
Comment (tomflinda) [inline behavior_tests/src/user-defined-sycl-file-extension/do_test.py]: Pls add source with extension name ".cpp", for example foo.cpp
Then what extension name is used for foo.cpp when each of "--sycl-file-extension=sycl-cpp, --sycl-file-extension=dp-cpp, and --sycl-file-extension=cpp" is specified 
Also, what the default extension name is used for main.cu, foo.cpp when all these three extension option are not specified.

The case also shoudl cover header files, like foo.cuh, main.h.

**Review by tomflinda — CHANGES_REQUESTED** [review]
Pls Update the PR tile to include the PR number merged in SYCLomatic code repo, like:
```
 [SYCLomatic #1465] Add test to check the rewording of DPCT1026 (#560)
```
So that we can see the feature PR related to this test.

File: behavior_tests/src/user-defined-sycl-file-extension/do_test.py (hunk 1 above)
Comment (the-slow-one) [reply behavior_tests/src/user-defined-sycl-file-extension/do_test.py]: Added _src.cpp_, _src.h_, _cuda_src.cuh_.

The _.cpp_ and _.h_ remains same in all the cases supported by the option _-sycl-file-extension_.

The default extension name is _.dp.cpp_ and _dp.hpp_. There is already a test related to it.

**Review by tomflinda — APPROVED** [review]
LGTM

**Pull request merged**